### PR TITLE
signatures: add ReleasePullErrorSignature

### PR DESCRIPTION
**Pull request by vrutkovs**
This checks journal.logs (bootstrap's journal) for "Pull failed" string to find clusters which can;t pull release image (misconfigured proxy etc.)

Example ticket: https://issues.redhat.com/browse/AITRIAGE-3135:

```
h1. Flapping validations:
|| validation           || count                           ||
| machine-cidr-defined | This validation flapped 3 times |
| ntp-synced           | This validation flapped 3 times |

h1. Release image cannot be pulled:

h2. Release image cannot be pulled on 8e760742-4e16-dc22-a64c-012f510db7c5 (dev-oshift-master01.ocp-ai.bos01.corp.akadevqa.com)
```

### CODE REVIEW

**Comment by omertuc** [comment]
For known issues, we should probably associate them with a particular Jira ticket to aggregate all of them

**Comment by omertuc** [comment]
Why did our preparing-for-installation image pull validations didn't catch this?

**Comment by omertuc** [comment]
I would look into it in more details for those tickets. It's interesting to find out why it was missed.

And maybe add a sentence about this in the triage signature "This should have been caught by our image pull validations, but it wasn't, try figuring out why"

File: tools/add_triage_signature.py [hunk 1]
```diff
@@ -1593,6 +1593,45 @@ def _process_ticket(self, url, issue_key):
...
 
 
+class ReleasePullErrorSignature(Signature):
+    """
+    This signature finds tickets for clusters where release image cannot be pulled by bootstrap node
+    """
+
+    ERROR_PATTERN = re.compile(r"release-image-download\.sh\[.+\]: Pull failed")
+
+    def __init__(self, jira_client):
```
Comment (omertuc) [inline tools/add_triage_signature.py:1603]: Needs generic constructor like other sigs

File: tools/add_triage_signature.py [hunk 2]
```diff
@@ -1593,6 +1593,45 @@ def _process_ticket(self, url, issue_key):
...
+        super().__init__(jira_client, comment_identifying_string="h1. Release image cannot be pulled")
+
+    def _update_ticket(self, url, issue_key, should_update=False):
```
Comment (omertuc) [inline tools/add_triage_signature.py:1606]: parent function name changed and should_update should be removed

File: tools/add_triage_signature.py [hunk 3]
```diff
@@ -1593,6 +1593,45 @@ def _process_ticket(self, url, issue_key):
...
+        url = self._logs_url_to_api(url)
```
Comment (omertuc) [inline tools/add_triage_signature.py:1607]: line can be safely deleted

File: tools/add_triage_signature.py [hunk 4]
```diff
@@ -1593,6 +1593,45 @@ def _process_ticket(self, url, issue_key):
...
+                continue
+
+            if self.ERROR_PATTERN.findall(journal_logs):
+                report += dedent(
+                    f"""
+                h2. Release image cannot be pulled on {host_id} ({self._get_hostname(host)})"""
+                )
+
+        if len(report) != 0:
+            self._update_triaging_ticket(issue_key, report, should_update=should_update)
```
Comment (omertuc) [inline tools/add_triage_signature.py:1632]: this function no longer accepts any parameters other than `report`

File: tools/add_triage_signature.py [hunk 5]
```diff
@@ -1593,6 +1593,49 @@ def _process_ticket(self, url, issue_key):
...
+    """
+
+    ERROR_PATTERN = re.compile(r"release-image-download\.sh\[.+\]: Pull failed")
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(
+            *args,
+            **kwargs,
+            comment_identifying_string="h1. Release image cannot be pulled:",
+            label="release_pull_error",
```
Comment (omertuc) [inline tools/add_triage_signature.py:1608]: I think this only works for ErrorSignature not Signature? not sure

**Comment by vrutkovs** [comment]
Lets merge it and see if we can get more tickets to find out why validations didn't catch it
/hold cancel

**Pull request merged**